lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/entity/EntityPermissionDaoImplTest.java

```diff
@@ -0,0 +1,536 @@
+package org.sagebionetworks.repo.model.dbo.entity;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.sagebionetworks.repo.model.util.AccessControlListUtil.createResourceAccess;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.sagebionetworks.repo.model.ACCESS_TYPE;
+import org.sagebionetworks.repo.model.AccessControlList;
+import org.sagebionetworks.repo.model.AccessControlListDAO;
+import org.sagebionetworks.repo.model.DataType;
+import org.sagebionetworks.repo.model.AuthorizationConstants.BOOTSTRAP_PRINCIPAL;
+import org.sagebionetworks.repo.model.EntityType;
+import org.sagebionetworks.repo.model.Node;
+import org.sagebionetworks.repo.model.NodeDAO;
+import org.sagebionetworks.repo.model.ObjectType;
+import org.sagebionetworks.repo.model.UserGroup;
+import org.sagebionetworks.repo.model.UserGroupDAO;
+import org.sagebionetworks.repo.model.dbo.dao.DataTypeDao;
+import org.sagebionetworks.repo.model.helper.DoaObjectHelper;
+import org.sagebionetworks.repo.model.jdo.KeyFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.google.common.collect.Sets;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(locations = { "classpath:jdomodels-test-context.xml" })
+public class EntityPermissionDaoImplTest {
+
+	@Autowired
+	private UserGroupDAO userGroupDAO;
+
+	@Autowired
+	private NodeDAO nodeDao;
+
+	@Autowired
+	AccessControlListDAO aclDao;
+
+	@Autowired
+	DataTypeDao dataTypeDao;
+
+	@Autowired
+	private DoaObjectHelper<Node> nodeDaoHelper;
+
+	@Autowired
+	private DoaObjectHelper<UserGroup> userGroupHelpler;
+
+	@Autowired
+	private DoaObjectHelper<AccessControlList> aclHelper;
+
+	@Autowired
+	EntityPermissionDao entityPermissionDao;
+
+	Long userOneId;
+	Long userTwoId;
+	Long userThreeId;
+	Long teamOneId;
+
+	Set<Long> userOneGroups;
+
+	Node project;
+	Node folder;
+	Node file;
+
+	Long projectId;
+	Long folderId;
+	Long fileId;
+
+	@BeforeEach
+	public void before() throws Exception {
```

Comment:
Additional test cases to include:

- acl with same id and but different object type
- two files share an acl
- add permission types that are not used